ddtrace/appsec/iast/_ast/ast_patching.py

```diff
@@ -56,32 +58,35 @@ def visit_ast(
     return modified_ast
 
 
+def astpatch_module(module):  # type: (ModuleType) -> Tuple[str, str]
+    return astpatch_source(module.__name__, origin(module))
+
+
 def astpatch_source(
     module_name,  # type: str
-    module_path="",  # type: str
+    module_path,  # type: Optional[str]
 ):  # type: (...) -> Tuple[str, str]
-
     if not module_path and not module_name:
         log.debug("astpatch_source called with no module path or name")
         return "", ""
 
     detected_module_path = module_path
     if not detected_module_path:
         # Get the module path from the module name (foo.bar -> foo/bar.py)
-        loader = pkgutil.get_loader(module_name)
+        module_watchdog = ModuleWatchdog._instance
 
-        assert loader
-        if hasattr(loader, "module_spec"):
-            detected_module_path = loader.module_spec.origin
+        assert module_watchdog
+        detected_module_spec = module_watchdog.find_spec(module_name)
+        if detected_module_spec and hasattr(detected_module_spec, "origin"):
+            detected_module_path = detected_module_spec.origin
```

Comment:
is this still required? Is `astpatch_source` called in other ways other than via the new `astpatch_module` helper, which is already passing in the module origin?

Reply:
Refactored and removed!